### PR TITLE
pkg/trace/obfuscate: fix bug where obfuscation fails for autovacuum sql text

### PR DESCRIPTION
**Pull request by jmeunier28**
### What does this PR do?

we are seeing query text like `autovacuum: VACUUM ANALYZE fake.table` fail obfuscation in the postgres integration check, and we want to be able to ingest things like this for DBM. It is failing due to the fact that it has a `:` followed by a space char. Because it has a `:`, the obfuscater assumes it is the beginning of a [bind variable]( https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql_tokenizer.go#L221-L228). Since the bindVar scanner code does not handle [this case](https://github.com/DataDog/datadog-agent/blob/ec0b667ba1423b8b466be30b7aef1cc684733d79/pkg/trace/obfuscate/sql_tokenizer.go#L556-L569 ), obfuscation fails. 

simply checking if the char in front of the `:` is a space will prevent this from happening

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by dujuku — CHANGES_REQUESTED** [review]

File: pkg/trace/obfuscate/sql_test.go [hunk 1]
```diff
@@ -83,6 +83,30 @@ func TestKeepSQLAlias(t *testing.T) {
 	})
 }
 
+func TestCanObfuscateAutoVacuum(t *testing.T) {
+	assert := assert.New(t)
+	for _, tt := range []struct{ in, out string }{
+		{
+			in:  "autovacuum: VACUUM ANALYZE public.monitor",
+			out: "autovacuum :  VACUUM ANALYZE public.monitor",
```
[the remark this answers is not in the document]
Comment (jmeunier28) [reply pkg/trace/obfuscate/sql_test.go]: OK, @justiniso it looks like I can't get around this very easily. Here is the flow of the current code:

1. We get to the `:` char and call [`tokenizer.Scan`](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L344)
2. We hit [this case](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql_tokenizer.go#L221-L229), and then (with my fix) we would return the current token, which is `:`
3. We come out of the Scan func, and then [hit this code](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L366-L379), which will write an extra `'  '` char after the `:`. Since we do not have have access the **next token** we can't know if this is something we should add a space to or not. 

I think we can "fix" this by creating a special `TokenKind` (like `Vacuum`? or _something_), and then add a [case here](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql.go#L368) to skip the space if we see this token. 🤷  seems kinda meh. 

More limitations of using the token parser I guess

[the remark this answers is not in the document]
Comment (jmeunier28) [reply pkg/trace/obfuscate/sql_test.go]: Talked about this offline, and decided we are OK with this behavior for now

File: pkg/trace/obfuscate/sql_tokenizer.go [hunk 2]
```diff
@@ -559,7 +559,7 @@ func (tkn *SQLTokenizer) scanBindVar() (TokenKind, []byte) {
 		token = ListArg
 		tkn.advance()
 	}
-	if !isLetter(tkn.lastChar) {
+	if !isLetter(tkn.lastChar) && !unicode.IsSpace(tkn.lastChar) {
```
Comment (justiniso) [inline pkg/trace/obfuscate/sql_tokenizer.go:562]: Technically, the token is neither a `ValueArg` nor `ListArg`; it is simply `':'`. If we end up applying a rule later which processes this token, then we should return the correct token.

File: pkg/trace/obfuscate/sql_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (justiniso) [reply pkg/trace/obfuscate/sql_test.go]: Hey @jmeunier28 after considering some of the edge cases like `@runtot:=0` in the tests, we have to do one of two things:

1. apply grammar in the tokenizer (conditional logic that looks at the next token and checks if it is a space). This would be really bad and could introduce a lot of nasty bugs.
2. add a rule in our "lexer" (i.e. the `tokenFilter` in `sql.go`) to concat previous tokens to single `:`. This is less bad, but more complicated and requires expanding the functionality of the lexer a bit more to support backtracking or peeking forward.

I think this case is small enough that it won't be an issue. We can apply #2 later, but the important thing is fixing the error so these queries properly obfuscate.

**Review by dujuku — CHANGES_REQUESTED** [review]
I agree with this direction, but one small comment about returning the correct token and not a bind var.

File: pkg/trace/obfuscate/sql_tokenizer.go (hunk 2 above)
Comment (jmeunier28) [reply pkg/trace/obfuscate/sql_tokenizer.go:562]: ah yes good point, let me fix that real quick

**Review by dujuku — APPROVED** [review]
👍

File: pkg/trace/obfuscate/sql_tokenizer.go [hunk 3]
```diff
@@ -223,6 +223,9 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 				tkn.advance()
 				return ColonCast, []byte("::")
 			}
+			if unicode.IsSpace(tkn.lastChar) {
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql_tokenizer.go:226]: Is it ever possible for the space to be missing after `autovacuum:`? I assume the answer is negative but asking just to be 100% sure.

File: pkg/trace/obfuscate/sql_tokenizer.go [hunk 4]
```diff
@@ -223,6 +223,9 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
...
+				return TokenKind(ch), tkn.bytes()
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql_tokenizer.go:227]:
```suggestion
				// example scenario: "autovacuum: VACUUM ANALYZE fake.table"
				return TokenKind(ch), tkn.bytes()
```
Just to be clear here. This code is fairly complex so we have to make sure it can be figured out without excessive digging around git blaming and PRs. In this case it's hard because we don't have a specific TokenKind, so the comment helps.

**Comment by gbbr** [comment]
You should add a release note for this.

File: pkg/trace/obfuscate/sql_tokenizer.go (hunk 3 above)
Comment (jmeunier28) [reply pkg/trace/obfuscate/sql_tokenizer.go:226]: yeah, I can't find any documentation that contradicts this

**Review by gbbr — APPROVED** [review]
Looks good! Thanks!
